pkg/aks/create_test.go

```diff
@@ -158,6 +158,27 @@ var _ = Describe("newManagedCluster", func() {
 		Expect(managedCluster.NetworkProfile.LoadBalancerSku).To(Equal(containerservice.Basic))
 	})
 
+	It("should successfully create managed cluster with custom network plugin no network profile", func() {
```

Comment:
the 2 unit tests are completely identical, please make sure one test checks that proper values are set when kubenet is chosen as network plugin and the second that network profile values are nil when are not set on cluster spec.

Reply:
Not sure such a test case is possible, because Azure and Kubenet plugin requires network profile during cluster creation see [here](https://github.com/rancher/aks-operator/blob/e0fecc7b505f148855ae47c44a5efc70f3fe5fc5/pkg/aks/create.go#L104)

Test cases are not identical, because first [test](https://github.com/rancher/aks-operator/blob/e0fecc7b505f148855ae47c44a5efc70f3fe5fc5/pkg/aks/create_test.go#LL168-L172) sets network profile values to empty or null. 


Second [test](https://github.com/rancher/aks-operator/blob/e0fecc7b505f148855ae47c44a5efc70f3fe5fc5/pkg/aks/create_test.go#L161) is using the values from [clusterSpec](https://github.com/rancher/aks-operator/blob/e0fecc7b505f148855ae47c44a5efc70f3fe5fc5/pkg/aks/create_test.go#LL448-L451)